bugtool/cmd/root.go

```diff
@@ -65,26 +66,42 @@ var (
 )
 
 func init() {
+	BugtoolRootCmd.Flags().BoolVar(&archive, "archive", true, "Create archive")
 	BugtoolRootCmd.Flags().BoolVar(&serve, "serve", false, "Start HTTP server to serve static files")
+	BugtoolRootCmd.Flags().BoolVar(&k8s, "k8s-mode", false, "Require Kubernetes pods to be found or fail")
 	BugtoolRootCmd.Flags().IntVarP(&port, "port", "p", 4444, "Port to use for the HTTP server, (default 4444)")
 	BugtoolRootCmd.Flags().StringVarP(&dumpPath, "tmp", "t", "/tmp", "Path to store extracted files")
 	BugtoolRootCmd.Flags().StringVarP(&host, "host", "H", "", "URI to server-side API")
 	BugtoolRootCmd.Flags().StringVarP(&k8sNamespace, "k8s-namespace", "", "kube-system", "Kubernetes namespace for Cilium pod")
 	BugtoolRootCmd.Flags().StringVarP(&k8sLabel, "k8s-label", "", "k8s-app=cilium", "Kubernetes label for Cilium pod")
 }
 
-func runTool() {
-	// Search for a Cilium pod and if one is found prefix all of the
-	// commands to use the pod(s).
+func setup() {
```

Comment:
This might be clearer if named `getVerifyCiliumPods`. I would also prefer not to change the global variable directly but to return the list and set it in runTool, but I suspect we already change global variables all over.

Reply:
Sure